Makefile

```diff
@@ -17,14 +17,16 @@ SO = so
 INSTALL += -D
 endif
 
-all: libsfark.$(SO)
+all: libsfark.$(SO).0.0.0
 
 clean:
-	-rm *.o libsfark.$(SO)
+	-rm *.o libsfark.$(SO).0.0.0
 
-libsfark.$(SO): $(OBJECTS)
-	$(CXX) -shared $(LDFLAGS) $(OBJECTS) -o libsfark.$(SO)
+libsfark.$(SO).0.0.0: $(OBJECTS)
+	$(CXX) -shared $(LDFLAGS) -Wl,-soname,libsfark.$(SO).0 $(OBJECTS) -o libsfark.$(SO).0.0.0 -lz
```

Comment:
Wouldn't this cause trouble when an application using libz links to another version of it?

Reply:
Less so than leaving it out. The same major version will work anyway, and recompiling things so that the issues remaining after this go away is a distribution/package manager task.